vllm/forward_context.py

```diff
@@ -26,10 +26,29 @@
 batchsize_forward_time: defaultdict = defaultdict(list)
 
 
+def _compute_chunked_local_num_tokens(num_tokens_across_dp_cpu: list[int],
+                                      max_num_tokens: int,
+                                      chunk_idx: int) -> list[int]:
+    dp_size = len(num_tokens_across_dp_cpu)
+    remaining = num_tokens_across_dp_cpu.copy()
+
+    for _ in range(chunk_idx + 1):
+        local_sizes = []
+        for i in range(dp_size):
+            to_process = min(remaining[i], max_num_tokens)
+            if to_process == 0:
+                to_process = 1  # ensure lockstep even if done
+            else:
+                remaining[i] -= to_process
+            local_sizes.append(to_process)
+    return local_sizes
```

Comment:
Is this right? We set `local_sizes = []` each iteration inside of the loop but then return it at the end of the function.
Could you double check?

Also instead of using python for loops, it might be better to use numpy operations.